libraries/chain/block_log.cpp

```diff
@@ -56,6 +58,18 @@ namespace eosio { namespace chain {
             void reopen();
 
             void close() {
+               //for a pruned log that has at least one block, see if we should vacuum it
+               //pruned_blocks == UINT32_MAX is kind of a hint to "please don't prune, but keep it a pruned log"
+               if( block_file.is_open() && index_file.is_open() && head && prune_blocks && *prune_blocks != UINT32_MAX) {
+                  const size_t first_data_pos = get_block_pos(first_block_num);
+                  block_file.seek_end(-sizeof(uint32_t));
+                  const size_t last_data_pos = block_file.tellp();
+                  if(last_data_pos - first_data_pos < 1024*1024*1024) {
```

Comment:
It was a nice round number to start discussion at. It’s not clear whether there is a preference when/if vacuuming should be performed automatically. I _thought_ I had a strong opinion on this initially but all the ongoing discussions make me uncertain now.

Clearly vacuuming is performed with the manual `eosio-blocklog --vacuum` command (block log is fortunate that it has this separate tool: ship has no such operation making this decision matrix a little tougher to decide on)

For automatic vacuuming there are then two opportunities:
* On exit, if the size to vacuum is small enough (what’s small enough? 1GB at the moment).
* On start, if the user no longer specifies a trim value. (the alternative here is that the log stays an “unvacuumed pruned log” but stops performing any more pruning until the user specifies otherwise in the future).

fwiw I've removed automatic vacuuming at the moment until we can come to agreement on the criteria for when it should happen.